packages/schematics/angular/migrations/update-11/update-dependencies.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { Rule } from '@angular-devkit/schematics';
+import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
+import {
+  addPackageJsonDependency,
+  getPackageJsonDependency,
+} from '../../utility/dependencies';
+import { latestVersions } from '../../utility/latest-versions';
+
+export default function (): Rule {
+  return (host, context) => {
+    const dependenciesToUpdate: Record<string, string> = {
+      '@types/jasmine': '~3.6.0',
+      'codelyzer': '^6.0.0',
```

Comment:
Are `codelyzer`, `protractor`, `ng-packagr` and `karma` needed since they are a peerDep of build-angular? AFAIK there are redundant.

Reply:
Without codelyzer the E2E tests fail: https://app.circleci.com/pipelines/github/angular/angular-cli/11428/workflows/5fb8edc6-b263-4cd7-9ac2-679b8c457298/jobs/206222

Reply:
Interesting I though adding it `packageGroup` was enough.

Reply:
Should we remove the other and leave codelyzer?

Reply:
removed the others